src/cmap/auth/mongodb_aws.ts

```diff
@@ -108,7 +108,7 @@ export class MongoDBAWS extends AuthProvider {
           return;
         }
 
-        if (serverNonce.compare(nonce, 0, nonce.length, 0, nonce.length) !== 0) {
+        if (!ByteUtils.equals(serverNonce.subarray(0, nonce.byteLength), nonce)) {
```

Comment:
This is interesting, I'm preserving what was here, but by using our local nonce length we are permitting the server to send us a nonce that's longer than the one we have in memory locally.

Reply:
I'm not sure it's possible within the confines of the auth protocol but if we send "abcd" and the server replies "abcdxyz" we won't enter this error handling because we only checked the section of the server response that was the length of what we sent to the server. But that is exactly the logic that was here before my change:

https://nodejs.org/dist/latest-v18.x/docs/api/buffer.html#bufcomparetarget-targetstart-targetend-sourcestart-sourceend

Reply:
should we file a follow up to fix this?

Reply:
filed!